utils/build/docker/python/fastapi/main.py

```diff
@@ -96,6 +101,71 @@ async def tag_value_post(tag_value: str, status_code: int, request: Request):
     return PlainTextResponse("Value tagged", status_code=status_code, headers=request.query_params)
 
 
+### BEGIN EXPLOIT PREVENTION
+
+
+@app.get("/rasp/lfi")
+@app.post("/rasp/lfi")
+async def rasp_lfi(request: Request):
+    file = None
+    if request.method == "GET":
+        file = request.query_params.get("file")
+    elif request.method == "POST":
+        body = await request.body()
+        try:
+            file = ((await request.form()) or json.loads(body) or {}).get("file")
+        except Exception as e:
+            print(repr(e), file=sys.stderr)
+        try:
+            if file is None:
+                file = xmltodict.parse(body).get("file")
+        except Exception as e:
+            print(repr(e), file=sys.stderr)
+            pass
+    if file is None:
+        return PlainTextResponse("missing file parameter", status_code=400)
+    try:
+        with open(file, "rb") as f_in:
+            f_in.seek(0, os.SEEK_END)
+            return PlainTextResponse(f"{file} open with {f_in.tell()} bytes")
+    except OSError as e:
+        return PlainTextResponse(f"{file} could not be open: {e!r}")
+
+
+@app.get("/rasp/ssrf")
+@app.post("/rasp/ssrf")
+def rasp_ssrf(request: Request):
+    print("rasp_ssrf", repr(request), file=sys.stderr)
+    domain = None
+    if request.method == "GET":
+        domain = request.query_params.get("domain")
+    elif request.method == "POST":
+        body = await request.body()
```

Comment:
This line causes this error trace on the startup:

```Traceback (most recent call last):
  File "/usr/local/bin/uvicorn", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 409, in main
    run(
  File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 575, in run
    server.run()
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/ddtrace/contrib/asyncio/patch.py", line 50, in traced_coro
    return await coro
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/usr/local/lib/python3.10/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/local/lib/python3.10/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "/usr/local/lib/python3.10/site-packages/ddtrace/internal/module.py", line 213, in _exec_module
    pre_exec_hook(self, module)
  File "/usr/local/lib/python3.10/site-packages/ddtrace/appsec/_iast/_loader.py", line 41, in _exec_iast_patched_module
    module_watchdog.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1017, in get_code
  File "<frozen importlib._bootstrap_external>", line 947, in source_to_code
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/main.py", line 143
    body = await request.body()
           ^^^^^^^^^^^^^^^^^^^^
SyntaxError: 'await' outside async function```

Reply:
fixed